specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -5189,6 +5203,45 @@
       },
       "description": "Response from a List SynonymMaps request. If successful, it includes the full definitions of all synonym maps."
     },
+    "EncryptionKey": {
+      "properties": {
+        "keyVaultKeyName": {
+          "type": "string",
+          "description": "The name of the KeyVault key."
+        },
+        "keyVaultKeyVersion": {
+          "type": "string",
+          "description": "The version of the KeyVault key."
+        },
+        "keyVaultUri": {
+          "type": "string",
+          "description": "The URI of the KeyVault."
+        },
+        "accessCredentials": {
+          "$ref": "#/definitions/AzureActiveDirectoryApplicationCredentials",
+          "description": "The credentials to use to authenticate to the provided KeyVault. Can be left unspecified if MSI (Managed Service Identity) is to be used instead."
+        }
+      },
+      "required": [
+        "keyVaultKeyName",
+        "keyVaultKeyVersion",
+        "keyVaultUri"
+      ],
+      "description": "The configuration to use encryption with customer managed keys."
+    },
+    "AzureActiveDirectoryApplicationCredentials": {
+      "properties": {
+        "applicationId": {
+          "type": "string",
+          "description": "The application id."
+        },
+        "applicationSecret": {
+          "type": "string",
+          "description": "The application secret."
+        }
+      },
+      "description" : "Credentials to be used to authenticate as an Azure Active Directory Application."
```

Comment:
Since we want to encourage the use of MSI, perhaps we should repeat the MSI comment also in this description

Reply:
Same comment as the one about search service date restriction. It seemed to make more sense to me to keep that kind of information directly where it's relevant (in this case, in the credential property of the encryptionKey class, where the user actually have to make a decision about whether or not to specify a value for that property or not.